lib/src/flutter_datawedge.dart

```diff
@@ -0,0 +1,228 @@
+          {String? commandIdentifier}) =>
+      _sendDataWedgeCommand(
+        DatawedgeApiTargets.getProfiles,
+        EmptyCommand.empty,
+        commandIdentifier: commandIdentifier,
+      );
+
+  /// This function for the result of the [requestProfiles] function and return it as a [List<String>]
+  /// [requestProfiles] has to be called before calling this function
+  Future<List<String>> waitForProfiles() {
+    final completer = Completer<List<String>>();
+    final subscription = onScannerEvent.listen((event) {
+      if (event.command == DatawedgeApiTargets.getProfiles.value) {
+        final profiles = event.resultInfo!['profiles'] as List<String>;
+        completer.complete(profiles);
+      }
+    });
+    return completer.future.whenComplete(() => subscription.cancel());
+  }
+
+  /// Gets the name of the Profile currently in use by DataWedge.
+  /// Answers using the [onScannerEvent] stream
+  /// [waitForActiveProfile] can be used to wait for the answer and returns a [String]
+  /// https://zebra-techdocs-archive.netlify.app/datawedge/13-0/guide/api/getactiveprofile/
+  Future<Result<void, FlutterDatawedgeException>> requestActiveProfile(
+          {String? commandIdentifier}) =>
+      _sendDataWedgeCommand(
+        DatawedgeApiTargets.getActiveProfile,
+        EmptyCommand.empty,
+        commandIdentifier: commandIdentifier,
+      );
+
+  /// This function for the result of the [requestActiveProfile] function and return it as a [String]
+  /// [requestActiveProfile] has to be called before calling this function
+  Future<String> waitForActiveProfile() {
+    final completer = Completer<String>();
+    final subscription = onScannerEvent.listen((event) {
+      if (event.command == DatawedgeApiTargets.getActiveProfile.value) {
+        final profile = event.resultInfo!['activeProfile'] as String;
+        completer.complete(profile);
+      }
+    });
+    return completer.future.whenComplete(() => subscription.cancel());
+  }
+
+  /// Create and configure a default Datawedge profile with the given name
+  /// Returns when the Command is executed NOT when DataWedge is ready to be operated again
+  /// For that use [onScannerEvent] to listen for the Result of the Command
+  Future<void> createDefaultProfile(
+      {required String profileName, String? commandIdentifier}) async {
+    final identifier = commandIdentifier ?? Uuid().v4();
+    _methodChannel.invokeMethod<void>(
+      MethodChannelMethods.createDataWedgeProfile.value,
+      jsonEncode({
+        "name": profileName,
+        'commandIdentifier': identifier,
+      }),
+    );
+  }
+
+  /// Returns the version of the Android OS
+  /// example: Android 4.4, Android 10
+  /// see also: https://developer.android.com/reference/android/os/Build.VERSION#RELEASE
+  Future<String?> platformVersion() => _methodChannel.invokeMethod<String>(
+        MethodChannelMethods.getPlatformVersion.value,
+      );
```

Comment:
Since we implement breaking changes, should we remove this? There are better packages for getting platform versions